moviepy/video/compositing/CompositeVideoClip.py

```diff
@@ -116,16 +116,21 @@ def __init__(
                 maskclips, self.size, is_mask=True, bg_color=0.0
             )
 
-        def make_frame(t):
-            """The clips playing at time `t` are blitted over one
-            another."""
+    def make_frame(self, t):
+        """The clips playing at time `t` are blitted over one another."""
+
+        frame = self.bg.get_frame(t).astype("uint8")
+        im = Image.fromarray(frame)
 
-            frame = self.bg.get_frame(t)
-            for clip in self.playing_clips(t):
-                frame = clip.blit_on(frame, t)
-            return frame
+        if self.bg.mask is not None:
+            frame_mask = self.bg.mask.get_frame(t)
+            im_mask = Image.fromarray(255 * frame_mask).convert("L")
+            im = im.putalpha(im_mask)
```

Comment:
There is a bug here : im.putalpha(im_mask) works in-place and returns None, so it should not be assigned back to im.